src/states/node.rs

```diff
@@ -1755,6 +1872,12 @@ impl Node {
         Ok(serialisation::deserialise(&serialised_connection_info)?)
     }
 
+    fn reset_rt_timer(&mut self) {
+        self.rt_msg_id = None;
```

Comment:
Yes, that's by design, so that the responses don't interfere with something that is currently happening. We will then send out a new request the next time the timer ticks.